mesonbuild/backend/ninjabackend.py

```diff
@@ -581,7 +581,7 @@ def detect_prefix(out):
 
         raise MesonException(f'Could not determine vs dep dependency prefix string. output: {stderr} {stdout}')
 
-    def generate(self, capture: bool = False, vslite_ctx: dict = None) -> T.Optional[dict]:
+    def generate(self, capture: bool = False, vslite_ctx: T.Optional[T.Dict] = None) -> T.Optional[T.Dict]:
```

Comment:
The typing around vslite_ctx is insane if you try and do the whole thing. Whoever added it and didn't add typing deserves a demerit.

Reply:
Yeah, I figured a little is better than none here

Reply:
#11049

🤷

Reply:
Originally it was:
```python
captured_compile_args_per_buildtype_and_target: dict = None
```

I completely lacked any interest in mucking around with this beyond defanging the name into `vslite_ctx` lol.

Reply:
@GertyP please come back and help us! :smile:

Reply:
This is a little difficult and awkard for me to say but after an initially encouraging start to my first contributions to meson (e.g. helpful, valuable, and encouraging discussions and feedback and actual progress on getting things merged), I've become quite turned off from the prospect of any further contributions, mostly because it seems there is a black hole in getting changes merged (I can understand and sympathise to some extent), but when that's in conjunction with a very limited or total lack of reasonable explanation or feedback on what is preventing any progress, despite repeated attempts to elicit even the slightest explanation over a long period of time, then eventually it starts to come across as a little unpleasant/unwelcoming, perhaps even unprofessional.

For example -
https://github.com/mesonbuild/meson/pull/12043

I have no complaint about how others initially engaged with feedback on this.  In fact, realising how it can sometimes be quite draining to give feedback through larger sets of changes, I tried to take care to give thanks for the feedback and time invested from both @tristan957 and @eli-schwartz.  However, it soon reached a point where all outstanding issues appeared to be resolved, yet there was no apparent progress in either merging or offering any objections or requests for further changes.

Not knowing what to make of this, I'd assume this would be a simple case of it being overlooked or temporarily de-prioritised by an overloaded set of owners/maintainers, so I'd repeatedly, politely ask for just any explanation but get nothing (although I could see that @tristan957 often tries to help and get things moving, for which I'm very grateful, but I guess your hands are tied to some extent).

I'd have been perfectly understanding with a response along the lines of, "Very busy.  Will eventually get round to it", or, "I'm not comfortable enough with this area of changes so we're trying to find someone more familiar with this area to look over and merge it", or, "I'm not sure about X/Y/Z and need to mull it over before I explain my problems/concerns".

Over several attempts and over a long time, trying elicit any explanation at all, without success, I think has to be down to one (or more) of a few possibilities -
- Too bottlenecked and cannot even be bothered to say so much as, "No objections but too busy with more ugent stuff until ..."
- Cannot be bothered to explain any further objections/requests, leaving me (and others) in the dark.
- Lack of suitable owners/reviewers in the area of these changes, yet cannot be bothered to say so, nor provide any update in improving this area (i.e. broadening the set of suitable owners/maintainers).
- Actively looking to dissuade me from any further contributions, for whatever reason, by way of being unhelpful/obstructive.
- Some kind of big misunderstanding, yet cannot be bothered to say what that is after multiple attempts to prompt for any explanation.

What would these possibilities say to you?  Would they indicate that this is something that is interested and welcoming of contributions and into which you should invest time; to learn, to develop your own knowledge and skills and to then contribute with fixes, features, reviews, and suggestions?

Sorry for the length of this reply, which isn't what you've asked for, but I feel diligence at least requires me to let you know why I've arrived at a position of being quite disillusioned and repelled from further contributing.

Reply:
Yes.  I hope I made it clear that you've been kind, helpful, and proactive and it should be clear to all that you're ultimately not in a position to give me the particular answers I'm after.

If, at any point, someone would have said, "We've far too many demands on far too few maintainers so please bear with us" then I'd have totally understood, done what I could to try to accommodate those difficulties, and perhaps even tried to help out in other ways... but the fact that something so easy to say was not said, despite me practically begging for someone to confirm this on several occasions, really left me wondering whether there were deeper problems at play.

In the absence of any of a number of  reasonable responses, it's then very difficult not to wonder whether there's anything like passive-agressive pettyness, unprofessionalism, dishonesty, close-mindedness, apathy, or organisational dysfunction going on, which does not encourage willingness to become further involved and invested.